### PR TITLE
fixes #904 Implement `--info`

### LINKED ISSUE #904: Implement `--info`
Some of our users really like the info we print about hardware and OS.

An idea: add `--info` console line argument which does not run the benchmarks, but simply prints the info.

```
BenchmarkDotNet=v0.11.1.786-nightly, OS=Windows 10.0.17134.285 (1803/April2018Update/Redstone4)
Intel Xeon CPU E5-1650 v4 3.60GHz, 1 CPU, 12 logical and 6 physical cores
Frequency=3507500 Hz, Resolution=285.1033 ns, Timer=TSC
.NET Core SDK=3.0.100-alpha1-009642
  [Host] : .NET Core 3.0.0-preview1-27004-04 (CoreCLR 4.6.27003.04, CoreFX 4.6.27003.02), 64bit RyuJIT
```

/cc @eerhardt

### PR DESCRIPTION
**Pull request by lahma**
* Adds HostEnvironmentInfo.GetInformation() to get the string presentation 
* Consolidates printing logic to summary

### CODE REVIEW

File: src/BenchmarkDotNet/Environments/HostEnvironmentInfo.cs [hunk 1]
```diff
@@ -109,5 +113,22 @@ public override IEnumerable<string> ToFormattedString()
         public bool IsDotNetCliInstalled() => !string.IsNullOrEmpty(DotNetSdkVersion.Value);
 
         private static string GetBenchmarkDotNetVersion() => BenchmarkDotNetInfo.FullVersion;
+
+        [PublicAPI]
+        public static string GetInformation()
+        {
+            var summary = new Summary(
```
Comment (AndreyAkinshin) [inline src/BenchmarkDotNet/Environments/HostEnvironmentInfo.cs]: I don't like an idea to create a fake summary to print information about the environment, it looks like a hack. We have to refactor it somehow.

File: src/BenchmarkDotNet/Exporters/AsciiDocExporter.cs [hunk 2]
```diff
@@ -17,11 +17,7 @@ private AsciiDocExporter()
         public override void ExportToLog(Summary summary, ILogger logger)
         {
             logger.WriteLine("....");
-            foreach (string infoLine in summary.HostEnvironmentInfo.ToFormattedString())
```
Comment (AndreyAkinshin) [inline src/BenchmarkDotNet/Exporters/AsciiDocExporter.cs]: This change will conflict with other PR: https://github.com/dotnet/BenchmarkDotNet/pull/893

**Comment by lahma** [comment]
@AndreyAkinshin thank you for the review! I reverted the conflicting changes and removed that refactoring. I changed the string building to use Summary's logic by providing minimal information required.

Waiting for your feedback and happy to tweak more where needed.

**Comment by AndreyAkinshin** [comment]
@lahma, now it looks much better, thanks! Could you also add information about this feature in the documentation?
@adamsitnik, could you take a look?

**Comment by lahma** [comment]
I've added info about the new switch to console-args.md.

**Comment by adamsitnik** [comment]
@lahma thanks a lot!

/cc @eerhardt

**Pull request merged**